src/Symfony/Component/Form/FormView.php

```diff
@@ -20,6 +20,8 @@ class FormView implements \ArrayAccess, \IteratorAggregate, \Countable
 {
     /**
      * The variables assigned to this view.
+     *
+     * @var array
```

Comment:
Why is it needed? We initialize it with an array, that's more than enough. I don't see the value of adding this annotation.

Reply:
@fabpot without the annotation `$this->vars` access in custom code triggers a psalm warning.

Reply:
Because the default value is an array. Adding the type is redundant.

Reply:
I agree with Fabien. The static analyzer should be fixed instead.

Reply:
Since the property is `public` a static code analyser cannot use the default value here since the type can change with any assignment made at runtime.

Reply:
@nicolas-grekas @fabpot `$this->vars = 42;` now this is not an array 🤷 

php is dynamically typed and there are currently no typed properties, it's impossible to infer type for properties.

Reply:
The code analyzer could decide to take the type of the default value when no annotation is provided.
This should be allowed on the tool side. Could be a configuration option if their devs don't agree it should be the default. But pushing for the rules of the tool to the code base is undesired to me.

Reply:
especially when it is a tool that is not part of our own CI tooling.

Reply:
@nicolas-grekas  static analys is not about "guessing", it's about verifying claims and statements

Reply:
As said, this is an array as determined by the default value, it won't be anything else.

Reply:
One of our goal is to minimize the noise created by phpdocs. So, when they are not needed, we avoid them.

Reply:
"it won't be anything else" --- it's impossible to infer it _reliably_

Reply:
And adding a phpdoc won't make it impossible to store an integer. Let's move on. We won't add it.

Reply:
Adding a phpdoc type would make it verifyable: you make a claim and you can statically verify it.